app/gilded-rose.ts

```diff
@@ -17,50 +17,90 @@ export class GildedRose {
     this.items = items;
   }
 
+  modifyQualityBy1(quality: number, mode: string) {
+    if (quality < 50 && mode == 'increase') {
+      quality = quality + 1;
+    }
+
+    if (quality > 0 && mode == 'decrease') {
+      quality = quality - 1;
+    }
+
+    return quality;
+  }
+
+  updateQualityForAgedBrie(item: Item) {
+    item.quality = this.modifyQualityBy1(item.quality, 'increase');
+  }
+
+  updateQualityForBackstage(item: Item) {
+    if (item.sellIn < 0) {
+      item.quality = 0;
+      return;
+    }
+
+    if (item.sellIn < 6) {
+      for (let i = 0; i < 3; i++) {
+        item.quality = this.modifyQualityBy1(item.quality, 'increase');
+      }
```

Comment:
instead of having this I would add a separate method

`modifyQualityBy(n, quality, mode)`  and call it in `modifyQualityBy1` with n = 1. Or you can also set the default value to 1